sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/MethodParameterLookup.cs

```diff
@@ -28,15 +28,27 @@ namespace SonarAnalyzer.Helpers
     // todo: this should come from the Roslyn API (https://github.com/dotnet/roslyn/issues/9)
     internal class MethodParameterLookup
     {
-        private readonly InvocationExpressionSyntax invocation;
+        private readonly ArgumentListSyntax argumentList;
         public IMethodSymbol MethodSymbol { get; }
 
         public MethodParameterLookup(InvocationExpressionSyntax invocation, SemanticModel semanticModel)

```

Comment:
You could call the constructor that accepts ArgumentListSyntax and SemanticModel from here. And perhaps remove the unused constructors...